src/__tests__/unit/controllers/hooks/router-store/test.tsx

```diff
@@ -0,0 +1,306 @@
+
+    let renderedFoo = 0;
+    const ComponentFoo = () => {
+      const [foo, setFoo] = useQueryParam('foo');
+      fooVal = foo;
+      fooUpdateFn = setFoo;
+      renderedFoo++;
+
+      return <div id="foo-count">{renderedFoo}</div>;
+    };
+    let renderedBar = 0;
+    const ComponentBar = () => {
+      const [bar, setBar] = useQueryParam('bar');
+      barVal = bar;
+      barUpdateFn = setBar;
+      renderedBar++;
+
+      return <div id="bar-count">{renderedBar}</div>;
+    };
+
+    const wrapper = mount(
+      <Router
+        routes={mockRoutes}
+        history={historyHelper.createBrowserHistory()}
+      >
+        <ComponentFoo />
+        <ComponentBar />
+      </Router>
+    );
+
+    expect(wrapper.find(ComponentFoo).html()).toEqual(
+      '<div id="foo-count">1</div>'
+    );
+    expect(wrapper.find(ComponentBar).html()).toEqual(
+      '<div id="bar-count">1</div>'
+    );
+
+    const { actions } = getRouterStore();
+
+    actions.push('/projects/123/board/456?foo=hello&bar=world');
+    await nextTick();
+
+    expect(fooVal).toEqual('hello');
+    expect(barVal).toEqual('world');
+    expect(wrapper.find(ComponentFoo).html()).toEqual(
+      '<div id="foo-count">2</div>'
+    );
+    expect(wrapper.find(ComponentBar).html()).toEqual(
+      '<div id="bar-count">2</div>'
+    );
+
+    act(() => fooUpdateFn('newVal'));
+    await nextTick();
+
+    // URL is now — /projects/123/board/456?foo=newVal&bar=world
+
+    expect(fooVal).toEqual('newVal');
+    expect(barVal).toEqual('world');
+    expect(wrapper.find(ComponentFoo).html()).toEqual(
+      '<div id="foo-count">3</div>'
+    );
+    expect(wrapper.find(ComponentBar).html()).toEqual(
+      '<div id="bar-count">2</div>'
+    );
+
+    act(() => barUpdateFn('newVal'));
+    await nextTick();
+
+    // URL is now — /projects/123/board/456?foo=newVal&bar=newVal
+
+    expect(fooVal).toEqual('newVal');
+    expect(barVal).toEqual('newVal');
+    expect(wrapper.find(ComponentFoo).html()).toEqual(
+      '<div id="foo-count">3</div>'
+    );
+    expect(wrapper.find(ComponentBar).html()).toEqual(
+      '<div id="bar-count">3</div>'
+    );
+  });
+});
+
+describe('usePathParam', () => {
+  beforeEach(() => {
+    history = historyHelper.createMemoryHistory(historyBuildOptions);
+    historyPushSpy = jest.spyOn(history, 'push');
+  });
+
+  afterEach(() => {
+    defaultRegistry.stores.clear();
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it('should return the right param value', () => {
+    let ppVal: string | undefined;
+
+    mount(
+      <Router routes={mockRoutes} history={history}>
+        <MockComponent>
+          {() => {
+            const [param] = usePathParam('projectId');
+            ppVal = param;
+
+            return <div>I am a subscriber</div>;
+          }}
+        </MockComponent>
+      </Router>
+    );
+    expect(ppVal).toEqual('123');
+  });
+
+  it('should return undefined for non-existent params', () => {
```

Comment:
could you add a test that updates more than 1 path param? i.e. `projectId` and `boardId` We should test that only the intended param was updated

Reply:
`usePathParam('key')` hook takes only one param at a time. There's no API way of updating multiple path params at the same time. @jackbrown

Reply:
Ah i meant use 2 different hooks in a single test. It's no biggie though I'm happy to leave it for now